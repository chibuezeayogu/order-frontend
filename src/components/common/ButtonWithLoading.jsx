import React from 'react'

const ButtonWithLoading = ({ text, loading }) => {
  return (
    <button
      className="form-control btn rounded submit px-3 btn-loading"
      type="submit"
      disabled={loading}
    >
      {loading ? (
        <>
          <span
            className="spinner-border spinner-border-sm"
            role="status"
            aria-hidden="true"
          />
          Loading...
        </>
      ) : (
        text
      )}
    </button>
  )
}

export default ButtonWithLoading
