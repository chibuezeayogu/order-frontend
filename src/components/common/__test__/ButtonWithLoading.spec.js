import React from 'react'
import { shallow } from 'enzyme'
import ButtonWithLoading from '../ButtonWithLoading'

const setUp = loading => {
  const props = {
    text: 'testing',
    loading
  }

  return shallow(<ButtonWithLoading {...props} />)
}

describe('Loader Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = setUp(false)
  })

  it('Should render without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render button with text testing', () => {
    expect(wrapper.find('button').text()).toBe('testing')
  })

  it('Should render button with text loading...', () => {
    wrapper.setProps({ loading: true })

    expect(wrapper.find('button').text()).toBe('Loading...')
  })
})
