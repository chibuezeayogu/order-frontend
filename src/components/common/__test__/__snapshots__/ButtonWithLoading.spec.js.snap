// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Should render without crashing 1`] = `
<button
  className="form-control btn rounded submit px-3 btn-loading"
  disabled={false}
  type="submit"
>
  testing
</button>
`;
